src/frontend/data/stores/useLivePanelState/index.spec.ts

```diff
@@ -0,0 +1,278 @@
+import { LivePanelDetail, useLivePanelState } from '.';
+
+describe('useLivePanelState()', () => {
+  it('init with default values', () => {
+    const state = useLivePanelState.getState();
+
+    expect(state.availableDetails).toEqual([]);
+    expect(state.currentDetail).toEqual(undefined);
+    expect(state.isPanelVisible).toBeFalsy();
+  });
+});
+
+describe('state.setAvailableDetails()', () => {
+  it('without changing available details has no side effect', () => {
+    useLivePanelState.setState({
+      isPanelVisible: true,
+      currentDetail: LivePanelDetail.CHAT,
+      availableDetails: [LivePanelDetail.APPLICATION, LivePanelDetail.CHAT],
+    });
+
+    let state = useLivePanelState.getState();
```

Comment:
IMO you should not store the state but directly use `useLivePanelState.getState()` everywhere.

```
    //  do not change the state
    useLivePanelState.getState().setAvailableDetails([
      LivePanelDetail.APPLICATION,
      LivePanelDetail.CHAT,
    ]);
```

and so on in your all file.